### ISSUE TITLE
fluid.layers.softmax_with_cross_entropy，ignore_index参数，实现存在问题

### ISSUE THREAD

**Issue by jianzfb** [PaddlePaddle/Paddle#25573]
PaddlePaddle 版本 1.8.2
GPU：P40
单卡环境

问题函数：
    loss = \
        fluid.layers.softmax_with_cross_entropy(logits,
                                                labels,
                                                ignore_index=255)
当前模型是一个两类的分割模型，logits来自模型获得每个位置的预测logits，形状N x 2；label是每个位置的真实标签值（0，1，255），形状 N x 1。其中255表示此位置为无效位置。原本期望通过设置ignore_index=255，来自动完成摒弃损失函数中对于这些无效位置的考虑（pytorch下是可以的）。
但是此时paddle的训练过程便无法收敛，损失函数值乱跳。

**Comment by baiyfbupt** [comment]
https://github.com/PaddlePaddle/PaddleSeg/blob/c2bc76f8336a1deef08e72411288c02dd737d389/pdseg/loss.py#L37 看到PaddleSeg在分割任务上也用了这个op，而且在多个任务上验证过，麻烦先确认下用法是否正确呢？

**Comment by jianzfb** [comment]
> https://github.com/PaddlePaddle/PaddleSeg/blob/c2bc76f8336a1deef08e72411288c02dd737d389/pdseg/loss.py#L37 看到PaddleSeg在分割任务上也用了这个op，而且在多个任务上验证过，麻烦先确认下用法是否正确呢？

是的 我就是在弄PaddleSeg，不过在调用此fluid.layers.softmax_with_cross_entropy函数前，已经通过调用，
    label = fluid.layers.elementwise_min(
        label, fluid.layers.assign(np.array([num_classes - 1], dtype=np.int32)))
将想要去掉的标签，除去了。也就是说，无法通过这样的操作来验证softmax_with_cross_entropy里面,ignore_index参数正确。

我这里 验证了，如果之前不通过elementwise_min，去除无效标签。期望通过softmax_with_cross_entropy里的ignore_index消除，是不可以的。
下图是，你们PaddleSeg里的softmaxloss损失函数，可以看一下，无效标签其实在进入softmax_with_cross_entropy前已经被消除了。
<img width="589" alt="截屏2020-07-17 上午11 13 50" src="https://user-images.githubusercontent.com/16849192/87744679-a91c4e00-c81e-11ea-934f-9780341622e5.png">

**Comment by phlrain** [comment]
https://github.com/PaddlePaddle/Paddle/pull/25549
这个pr已经修复了问题，刚merge进去develop分支，正式版本会在1月左右发布
